### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by hortega**

### CODE REVIEW

File: WarmTransfer.Web.Tests/Domain/CallCreatorTest.cs [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
+        }
+
+        private static Mock<ITwilioRestClient> SetupTwilioRestClientMock(Request request)
+        {
+            var mockClient = new Mock<ITwilioRestClient>();
+            
+            mockClient
+                .Setup(c => c.RequestAsync(request))
+                .Returns(Task.FromResult(new Response(System.Net.HttpStatusCode.OK, "")));
+            mockClient
```
Comment (joliveros) [inline WarmTransfer.Web.Tests/Domain/CallCreatorTest.cs]: I would split these blocks of code with a blank line for readability

File: WarmTransfer.Web.Tests/app.config [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <add key="TwilioAccountSid" value="account-sid" />
     <add key="TwilioAuthToken" value="auth-token" />
     <add key="TwilioPhoneNumber" value="twilio-phone-number" />
-    <add key="Domain" value="example.com"/>
```
Comment (joliveros) [inline WarmTransfer.Web.Tests/app.config]: You could probably get rid of this in the settings by doing something like this when building the url
https://github.com/TwilioDevEd/clicktocall-csharp/blob/next-gen/ClickToCall.Web/Controllers/CallCenterController.cs#L61-L62

File: WarmTransfer.Web/Domain/TwiMLGenerator.cs [hunk 3]
```diff
@@ -5,20 +5,28 @@ namespace WarmTransfer.Web.Domain
...
         }
 
-        public static TwilioResponse GenerateWait()
+        public static string GenerateWait()
         {
-            return new TwilioResponse()
-                .Say("Thank you for calling. Please wait in line for a few seconds. An agent will be with you shortly.")
-                .Play("http://com.twilio.music.classical.s3.amazonaws.com/BusyStrings.mp3");
+            return new VoiceResponse()
+                .Say("Thank you for calling. Please wait in line for a few seconds. " +
```
Comment (joliveros) [inline WarmTransfer.Web/Domain/TwiMLGenerator.cs]: nice +1

**Comment by joliveros** [comment]
LGTM

RTGM!

**Comment by hortega** [comment]
thanks for the review @joliveros

File: WarmTransfer.Web/Domain/TwilioCapability.cs [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
         {
-            var twilioCapability = new TwilioCapability(Config.AccountSid, Config.AuthToken);
-
-            twilioCapability.AllowClientIncoming(agentId);
-            return twilioCapability.GenerateToken();
+            var scopes = new HashSet<IScope>
+            {
+                { new IncomingClientScope(agentId) }
+            };
+            var lol = Config.AccountSid;
```
Comment (well1791) [inline WarmTransfer.Web/Domain/TwilioCapability.cs]: @smendes lol 😂

File: WarmTransfer.Web/Domain/TwilioCapability.cs [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
+            var clientCapibility = new ClientCapability(Config.AccountSid, Config.AuthToken, scopes: scopes);
+            return clientCapibility.ToJwt();
+            
+            //twilioCapability.AllowClientIncoming(agentId);
+            //return twilioCapability.GenerateToken();
```
Comment (well1791) [inline WarmTransfer.Web/Domain/TwilioCapability.cs]: @smendes no need for this comments! :+1:

**Pull request merged**